app/frontend/src/features/messages/groupchats/GroupChatListItem.tsx

```diff
@@ -1,31 +1,99 @@
-import { Box, BoxProps, Typography } from "@material-ui/core";
+import {
+  ListItem,
+  ListItemAvatar,
+  ListItemProps,
+  ListItemText,
+} from "@material-ui/core";
 import { makeStyles } from "@material-ui/core/styles";
+import { Skeleton } from "@material-ui/lab";
 import React from "react";
+import Avatar from "../../../components/Avatar";
 import { GroupChat } from "../../../pb/conversations_pb";
-import ControlMessageView from "../messagelist/ControlMessageView";
-import MessageView from "../messagelist/MessageView";
-import { isControlMessage } from "../utils";
+import { firstName } from "../../../utils/names";
+import { useAuthContext } from "../../auth/AuthProvider";
+import useUsers from "../../userQueries/useUsers";
+import {
+  controlMessageText,
+  groupChatTitleText,
+  isControlMessage,
+  messageTargetId,
+} from "../utils";
+import classnames from "classnames";
 
 const useStyles = makeStyles({ root: {} });
 
-export interface GroupChatListItemProps extends BoxProps {
+export interface GroupChatListItemProps extends ListItemProps {
   groupChat: GroupChat.AsObject;
 }
 
 export default function GroupChatListItem({
   groupChat,
+  className,
 }: GroupChatListItemProps) {
   const classes = useStyles();
+  const currentUserId = useAuthContext().authState.userId!;
+  const latestMessageAuthorId = groupChat.latestMessage?.authorUserId;
+
+  const groupChatMembersQuery = useUsers(groupChat.memberUserIdsList);
+
+  //the avatar is of the latest message author (if it's not the logged in user),
+  //otherwise any user that's not the logged in user, otherwise logged in user
+  const avatarUserId =
+    latestMessageAuthorId !== null && latestMessageAuthorId !== currentUserId
+      ? latestMessageAuthorId
+      : groupChat.memberUserIdsList.find((id) => id !== currentUserId) ??
+        currentUserId;
+  //title is the chat title, or all the member's names except current user joined together
+  const title = groupChatTitleText(
+    groupChat,
+    groupChatMembersQuery,
+    currentUserId
+  );
+  //text is the control message text or message text
+  let text = "";
+  if (groupChat.latestMessage) {
+    const authorName =
+      firstName(
+        groupChatMembersQuery.data?.get(groupChat.latestMessage?.authorUserId)
+          ?.name
+      ) || "";
+    if (isControlMessage(groupChat.latestMessage)) {
+      const targetName = groupChat.latestMessage
+        ? firstName(
+            groupChatMembersQuery.data?.get(
+              messageTargetId(groupChat.latestMessage)
+            )?.name
+          ) || ""
+        : undefined;
```

Comment:
```typescript
const targetName = firstName(
  groupChatMembersQuery.data?.get(
    messageTargetId(groupChat.latestMessage)
  )?.name
) || ""
```
We could simplify to that instead since there's already a check for `groupChat.latestMessage` above so we know it's truthy here.